src/MCPClient/lib/clientScripts/sanitize_names.py

```diff
@@ -21,100 +21,74 @@
 # @subpackage MCPClient
 # @author Joseph Perry <joseph@artefactual.com>
 
-import string
 import os
-from shutil import move as rename
-import sys
-import unicodedata
+import re
+import shutil
+
+from scandir import scandir
 from unidecode import unidecode
-from archivematicaFunctions import unicodeToStr
 
-VERSION = "1.10." + "$Id$".split(" ")[1]
-valid = "-_.()" + string.ascii_letters + string.digits
-replacementChar = "_"
+from archivematicaFunctions import strToUnicode
 
+VERSION = "1.10." + "$Id$".split(" ")[1]
 
-def transliterate(basename):
-    # We get a more meaningful name sanitization if UTF-8 names
-    # are correctly decoded to unistrings instead of str
-    try:
-        return unidecode(basename.decode("utf-8"))
-    except UnicodeDecodeError:
-        return unidecode(basename)
+# Letters, digits and a few punctuation characters
+ALLOWED_CHARS = re.compile(r"[^a-zA-Z0-9\-_.\(\)]")
+REPLACEMENT_CHAR = "_"
 
 
-def sanitizeName(basename):
-    ret = ""
-    basename = transliterate(basename)
-    for c in basename:
-        if c in valid:
-            ret += c
-        else:
-            ret += replacementChar
-    return ret.encode("utf-8")
+def sanitize_name(basename):
+    unicode_name = unidecode(strToUnicode(basename))
+    # Handle the case where '' is returned by unidecode
+    if unicode_name == "":
+        unicode_name = strToUnicode(basename)
+    return ALLOWED_CHARS.sub(u"_", unicode_name)
```

Comment:
Should this be

```suggestion
    return ALLOWED_CHARS.sub(REPLACEMENT_CHAR, unicode_name)
```

?